src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,3168 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
+                .ForIndex(2, s =>
+                    s.SetPublicId(SqidEncoder.Encode(5)) // Indicator added
+                        .SetLabel("d"))
+                .ForIndex(3, s =>
+                    s.SetPublicId(SqidEncoder.Encode(6)) // Indicator added
+                        .SetLabel("c"))
+                .GenerateList(4);
+
+            var (newVersion, instanceId) = await CreateDataSetNextVersion(
+                originalVersion: originalVersion,
+                indicatorMeta: newIndicatorMeta);
+
+            await CreateChanges(instanceId);
+
+            var indicatorMetaChanges = await GetIndicatorMetaChanges(newVersion);
+
+            // 6 Indicator changes
+            Assert.Equal(6, indicatorMetaChanges.Count);
+            Assert.All(indicatorMetaChanges, c => Assert.Equal(newVersion.Id, c.DataSetVersionId));
+
+            var oldIndicatorMetas = originalVersion.IndicatorMetas
+                .ToDictionary(m => m.PublicId);
+
+            var newIndicatorMetas = newVersion.IndicatorMetas
+                .ToDictionary(m => m.PublicId);
+
+            // The changes should be inserted into each database table ordered alphabetically by 'Label'.
+            // They should also be ordered such that all additions come last.
+
+            // Therefore, the expected order of Indicator changes are (as per their Public IDs):
+            // Sqid index 2 deleted
+            // Sqid index 4 changed
+            // Sqid index 3 changed
+            // Sqid index 1 deleted
+            // Sqid index 6 added
+            // Sqid index 5 added
+
+            AssertIndicatorDeleted(
+                expectedIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(2)],
+                change: indicatorMetaChanges[0]);
+            AssertIndicatorChanged(
+                expectedOldIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(4)],
+                expectedNewIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(4)],
+                change: indicatorMetaChanges[1]);
+            AssertIndicatorChanged(
+                expectedOldIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(3)],
+                expectedNewIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(3)],
+                change: indicatorMetaChanges[2]);
+            AssertIndicatorDeleted(
+                expectedIndicatorMeta: oldIndicatorMetas[SqidEncoder.Encode(1)],
+                change: indicatorMetaChanges[3]);
+            AssertIndicatorAdded(
+                expectedIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(6)],
+                change: indicatorMetaChanges[4]);
+            AssertIndicatorAdded(
+                expectedIndicatorMeta: newIndicatorMetas[SqidEncoder.Encode(5)],
+                change: indicatorMetaChanges[5]);
+        }
+
+        private static void AssertSingleIndicatorDeleted(
+            IReadOnlyList<IndicatorMetaChange> changes,
+            IndicatorMeta expectedIndicatorMeta)
```

Comment:
Could simplify this parameter to just `expectedIndicator`. 

Similar thing can be done across all the assertion methods.